evcxr: init at 0.3.3

protoben opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

[evcxr](https://github.com/google/evcxr) is a Rust REPL. It's lacking some features, but it's relatively polished, and it's actively maintained. I'd like to see it in nixpkgs. 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
evcxr: init at 0.3.3

symphorien commented on pkgs/development/interpreters/evcxr/default.nix:
```diff
@@ -0,0 +1,28 @@
+  cargoPatches = [ ./cargo-lock.patch ];
+
+  nativeBuildInputs = [ pkgconfig zeromq ];
+  buildInputs = [ cargo makeWrapper ];
```
makeWrapper should be a `nativeBuildInput`.

protoben replied:
Indeed. Fixed.

symphorien commented on pkgs/development/interpreters/evcxr/default.nix:
```diff
@@ -0,0 +1,28 @@
+  cargoSha256 = "1pvw1jfi4p0cla4xxzgxh1jskvbmsyi4l5cimiis7093hyjs0svw";
+  cargoPatches = [ ./cargo-lock.patch ];
+
+  nativeBuildInputs = [ pkgconfig zeromq ];
```
zeromq is likely to be a buildInput.

protoben replied:
zeromq was only needed by the Jupyter kernel. I figure that should probably be its own package, so I've removed it from the build, which eliminates the dependency.

Commit:
evcxr: makeWrapper -> nativeBuildInputs and omit Jupyter kernel

The Jupyter kernel should be its own package. Omitting it and its dependencies here.

Ma27 commented:
@GrahamcOfBorg build evcxr

Ma27 committed:
evcxr: fix hash and `cc` access

The `cargoSha256` hashes across the sourcetree had to be altered after
the last `cargo-vendor` bump. Also ensured that `cc` is available in
`$PATH` to avoid startup errors in the REPL.

Ma27 commented:
Fixed the `cargoSha256` hash that needed to be altered after the last `cargo-vendor` bump. Also ensured that `cc` is always available by wrapping the binary.

@GrahamcOfBorg build evcxr

Ma27 committed:
evcxr: fix darwin build

It seems as the `Security` framework is missing during the linking in a
darwin build environment:

https://logs.nix.ci/?key=nixos/nixpkgs.60675&attempt_id=42d9c159-f1a2-4abc-ab14-9dea23d941b1

Ma27 commented:
@GrahamcOfBorg build evcxr

Ma27 commented:
Built and tested locally, appears to work fine, thanks a lot for working on this!
